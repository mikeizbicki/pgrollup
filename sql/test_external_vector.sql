SET client_min_messages TO WARNING;

create table vector_test (
    id serial primary key,
    a vector(3)
);

insert into vector_test (a) VALUES ('[1,2,3]'), ('[2,3,4]'), (NULL);

CREATE MATERIALIZED VIEW vector_test_rollup1 AS (
    SELECT
        count(a),
        vector_sum(a),
        vector_avg(a),
        vector_sum(a)/count(a)
    FROM vector_test
);

insert into vector_test (a) VALUES ('[1,2,3]'), ('[2,3,4]'), (NULL);

select assert_rollup('vector_test_rollup1');

select * from vector_test_rollup1;

drop table vector_test cascade;
