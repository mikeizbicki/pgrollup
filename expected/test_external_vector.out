SET client_min_messages TO WARNING;
create table vector_test (
    id serial primary key,
    a vector(3)
);
insert into vector_test (a) VALUES ('[1,2,3]'), ('[2,3,4]'), (NULL);
CREATE MATERIALIZED VIEW vector_test_rollup1 AS (
    SELECT
        count(a),
        vector_sum(a),
        vector_avg(a),
        vector_sum(a)/count(a)
    FROM vector_test
);
insert into vector_test (a) VALUES ('[1,2,3]'), ('[2,3,4]'), (NULL);
select assert_rollup('vector_test_rollup1');
 assert_rollup 
---------------
 
(1 row)

select * from vector_test_rollup1;
 count | vector_sum |  vector_avg   | (vector_sum(vector_test.a)/(count(vector_test.a))::real) 
-------+------------+---------------+----------------------------------------------------------
     4 | [6,10,14]  | [1.5,2.5,3.5] | [1.5,2.5,3.5]
(1 row)

drop table vector_test cascade;
